src/client/src/models/Scene.js

```diff
@@ -112,20 +137,52 @@ const Scene = {
   // self-explanatory. Sets the currentScene property to the scene corresponding to the sceneId
   setCurrentScene: function(sceneId) {
     const list = Scene.getScenes()
-    Scene.currentScene = list[sceneId]
+    Scene.currentScene = list[Scene.findSceneIndex(sceneId)]
   },
 
   currentCaption: null,
 
   setCurrentCaption: function(captionId) {
-    Scene.currentCaption = Scene.currentScene.captions[captionId]
+    Scene.currentCaption = Scene.currentScene.captions[Scene.findCaptionIndex(captionId)]
+  },
+
+  findCaptionIndex: function (captionId) {
+    for (var i = 0; i < Scene.currentScene.captions.length; i++) {
+      if (Scene.currentScene.captions[i].id == captionId) {
+        return i
+      }
+    }
+
+    return -1
+  },
+
+  addCaption: function () {
+    var caption_index = 0
+    // Finds the max scene id in the list
+    if (Scene.currentScene.captions.length == 0) {
+      caption_index = 0
+    } else {
+      caption_index = 0
```

Comment:
`var` scopes either globally or locally, since this is in the function i don't think it's necessary to redeclare it to 0 here

Reply:
Yup and fixed it to reflect and new name called `caption_max_id`.